### PR TITLE
[tests-only][full-ci]Improve then steps in apiShareCreateSpecialToShares1 suite

### PR DESCRIPTION
**Pull request by sushmita56**
## Description
This PR adds ``` Then ``` steps to existing API test scenarios to better check the results of ``` When ``` steps.

### Suite Covered
- ```apiShareCreateSpecialToShare1```

## Related Issues
- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?
- Locally
- CI

## Types of changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by kiranparajuli589** [comment]
lint failed. try `make test-php-style`

File: tests/acceptance/features/apiShareCreateSpecialToShares1/createShareExpirationDate.feature [hunk 1]
```diff
@@ -34,18 +32,14 @@ Feature: a default expiration date can be specified for shares with users or gro
...
       | path        | /FOLDER    |
       | shareType   | user       |
       | shareWith   | Brian      |
       | permissions | read,share |
       | expireDate  | +15 days   |
-    Then the OCS status code should be "<ocs_status_code>"
-    And the HTTP status code should be "<http_status_code>"
     When user "Brian" accepts share "/FOLDER" offered by user "Alice" using the sharing API
-    Then the OCS status code should be "<ocs_status_code>"
-    And the HTTP status code should be "<http_status_code>"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiShareCreateSpecialToShares1/createShareExpirationDate.feature:47]: Aren't these necessary?

File: tests/acceptance/features/apiShareCreateSpecialToShares1/createShareExpirationDate.feature [hunk 2]
```diff
@@ -54,26 +48,22 @@ Feature: a default expiration date can be specified for shares with users or gro
...
```
Comment (SwikritiT) [inline tests/acceptance/features/apiShareCreateSpecialToShares1/createShareExpirationDate.feature:75]: Here too

File: tests/acceptance/features/apiShareCreateSpecialToShares1/createShareExpirationDate.feature [hunk 3]
```diff
@@ -747,20 +715,22 @@ Feature: a default expiration date can be specified for shares with users or gro
...
+    And user "Alice" has created a share with settings
       | path        | textfile0.txt |
       | shareType   | user          |
       | shareWith   | Brian         |
       | permissions | read,share    |
-    And user "Brian" accepts share "/textfile0.txt" offered by user "Alice" using the sharing API
-    Then the fields of the last response to user "Alice" should include
+    When user "Brian" accepts share "/textfile0.txt" offered by user "Alice" using the sharing API
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "<ocs_status_code>"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiShareCreateSpecialToShares1/createShareExpirationDate.feature:725]: Why are we adding these two steps in some and removing them in others? I'm a little bit confused

File: tests/acceptance/features/apiShareCreateSpecialToShares1/createShareExpirationDate.feature (hunk 1 above)
Comment (sushmita56) [reply tests/acceptance/features/apiShareCreateSpecialToShares1/createShareExpirationDate.feature:47]: `Then the info about the last share by user "Alice" with user "Brian" should include` step checks the info about the share and the status code both at once. So I removed the `Then` steps for asserting the status code from here.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by SagarGi — COMMENTED** [review]
lgtm :+1:

**Review by individual-it — APPROVED** [review]

**Pull request merged**